bodhi/tests/server/scripts/test_skopeo_lite.py

```diff
@@ -0,0 +1,728 @@
+
+                manifest = {
+                    'schemaVersion': 2,
+                    'manifests': [{
+                        'mediaType': MEDIA_TYPE_OCI,
+                        'size': len(manifest_bytes),
+                        'digest': manifest_digest,
+                        'platform': {
+                            'architecture': arch,
+                            'os': 'linux',
+                        }
+                    }]
+                }
+
+        manifest_bytes = to_bytes(json.dumps(manifest))
+        return self.add_manifest(name, tag, manifest_bytes)
+
+    def check_fake_image(self, name, tag, digest, content_type):
+        manifest_bytes = self.get_manifest(name, tag)
+        assert make_digest(manifest_bytes) == digest
+
+        manifest = json.loads(to_text(manifest_bytes))
+        if content_type in (MEDIA_TYPE_LIST_V2, MEDIA_TYPE_OCI_INDEX):
+            manifest_digest = manifest['manifests'][0]['digest']
+            manifest_bytes = self.get_manifest(name, manifest_digest)
+            manifest = json.loads(to_text(manifest_bytes))
+
+        config_digest = manifest['config']['digest']
+        assert make_digest(self.get_blob(name, config_digest)) == config_digest
+
+        layer_digest = manifest['layers'][0]['digest']
+        assert make_digest(self.get_blob(name, layer_digest)) == layer_digest
+
+
+@responses.activate
+@pytest.mark.parametrize('content_type',
+                         (MEDIA_TYPE_OCI, MEDIA_TYPE_OCI_INDEX,
+                          MEDIA_TYPE_MANIFEST_V2, MEDIA_TYPE_LIST_V2))
+def test_skopeo_copy_basic(content_type):
+    """
+    Test copying from one server to another
+    """
+    runner = testing.CliRunner()
+
+    reg1 = MockRegistry('registry1.example.com')
+    reg2 = MockRegistry('registry2.example.com')
+    digest = reg1.add_fake_image('repo1', 'latest', content_type)
+
+    result = runner.invoke(
+        skopeo_lite.copy,
+        ['docker:registry1.example.com/repo1:latest', 'docker:registry2.example.com/repo2:latest'],
+        catch_exceptions=False)
+
+    assert result.exit_code == 0
+
+    reg2.check_fake_image('repo2', 'latest', digest, content_type)
+
+
+@responses.activate
+@pytest.mark.parametrize('content_type',
+                         (MEDIA_TYPE_OCI, MEDIA_TYPE_OCI_INDEX,
+                          MEDIA_TYPE_MANIFEST_V2, MEDIA_TYPE_LIST_V2))
+def test_skopeo_copy_link(content_type):
+    """
+    Testing copying on the same server, avoiding download/upload
+    """
+    runner = testing.CliRunner()
+
+    reg1 = MockRegistry('registry1.example.com')
+    digest = reg1.add_fake_image('repo1', 'latest', content_type)
+
+    result = runner.invoke(
+        skopeo_lite.copy,
+        ['docker:registry1.example.com/repo1:latest', 'docker:registry1.example.com/repo2:latest'],
+        catch_exceptions=False)
+
+    assert result.exit_code == 0
+
+    reg1.check_fake_image('repo2', 'latest', digest, content_type)
+
+
+@responses.activate
```

Comment:
Neat, I hadn't seen this lib before. Looks nice! I'd been meaning to try out vcrpy, but I mostly have just used mock and that's not pretty.